main/src/main/java/org/mobilitydata/gtfsvalidator/cli/ValidateCliParameters.java

```diff
@@ -0,0 +1,28 @@
+package org.mobilitydata.gtfsvalidator.cli;
+
+import java.io.IOException;
+
+/**
+ * Provides convenient method to validate the requirement on CLI parameters
+ */
+public class ValidateCliParameters {
+    private Arguments args;
+
+    ValidateCliParameters(Arguments args) {
+        this.args = args;
+    }
+
+    /**
+     * Validate requirement on CLI parameters
+     * @throws IOException
+     */
+    public void validate() throws IOException {
```

Comment:
IMHO this really isn't an appropriate use of `IOException`. The validate method is intended to validate the input, and should return a value with the result of validation instead of throwing an exception to indicate the result of validate.

Reply:
Is that always true? I've seen some multithreading patterns were the runnable would either return a value OR an exception. Some always return an exception with a Result object in it, containing either a result or an error.
What would be the reason to not do something similar here and stay bare?

Reply:
One general reason to avoid it is that throwing exceptions is expensive - see https://www.baeldung.com/java-exceptions-performance.

Reply:
To elaborate further, IMHO exceptions should be used for unexpected things that happen during program execution, not for expected results. In this case, we have clear expectations for what is valid and what is not, and as a result it should be handled in normal control flow and not an exception.

I can't speak to whether this is always true - there are probably exceptions :).

Reply:
thks for the clarifications. I agree with you then :)